### PR TITLE
[build] SONiC platform daemons depend on swsscommon library

### PR DESCRIPTION
**Pull request by jleveque**
**- Why I did it**

To ensure proper build/installation of SONiC platform daemons (ledd, psud, syseepromd, thermalctld, xcvrd).

**- How I did it**

Add dependencies in appropriate makefiles

**- How to verify it**

**- Which release branch to backport (provide reason below if selected)**

- [ ] 201811
- [ ] 201911
- [ ] 202006

### CODE REVIEW

**Comment by jleveque** [comment]
@qiluo-msft: Check builds are failing because mock swsscommon calls appear to be unimplemented. Have you seen this before?

File: rules/sonic-psud.mk [hunk 1]
```diff
@@ -3,5 +3,6 @@
 SONIC_PSUD = sonic_psud-1.0-py2-none-any.whl
 $(SONIC_PSUD)_SRC_PATH = $(SRC_PATH)/sonic-platform-daemons/sonic-psud
 $(SONIC_PSUD)_DEPENDS = $(SONIC_PY_COMMON_PY2)
+$(SONIC_PUSD)_DEBS_DEPENDS = $(LIBSWSSCOMMON) $(PYTHON_SWSSCOMMON) $(PYTHON3_SWSSCOMMON)
```
Comment (qiluo-msft) [inline rules/sonic-psud.mk:6]: >SONIC_PUSD [](start = 2, length = 10)

PSUD

Comment (jleveque) [reply rules/sonic-psud.mk]: Thanks! Fixed.

**Comment by qiluo-msft** [comment]
I guess it is related to this part of dynamic behavior in test
```
try:
    from swsscommon import swsscommon
except ImportError as e:
    from tests import mock_swsscommon as swsscommon
```

It is not recommended practice. It assume swsscommon is missing in unit test environment.

---
In reply to: [727131567](https://github.com/Azure/sonic-buildimage/pull/5922#issuecomment-727131567) [](ancestors = 727131567)

**Comment by jleveque** [comment]
> I guess it is related to this part of dynamic behavior in test
> 
> ```
> try:
>     from swsscommon import swsscommon
> except ImportError as e:
>     from tests import mock_swsscommon as swsscommon
> ```
> 
> It is not recommended practice. It assumes swsscommon is missing in unit test environment.
> 
> In reply to: [727131567](https://github.com/Azure/sonic-buildimage/pull/5922#issuecomment-727131567) [](ancestors = 727131567)

Good find! Thanks, Qi! I have opened a PR which should fix this: https://github.com/Azure/sonic-platform-daemons/pull/112

**Comment by lguohan** [comment]
retest this please

**Review by lguohan — APPROVED** [review]

**Pull request merged**